frontend/src/metabase/containers/DataPicker/CardPicker/CardPickerContainer.tsx

```diff
@@ -0,0 +1,176 @@
+  CollectionsLoaderProps;
+
+function mapStateToProps(state: State) {
+  return {
+    currentUser: getUser(state),
+  };
+}
+
+function CardPickerContainer({
+  value,
+  collections,
+  rootCollection,
+  currentUser,
+  targetModel,
+  onChange,
+  onBack,
+}: CardPickerProps) {
+  const { schemaId } = value;
+
+  const [selectedCollectionId, setSelectedCollectionId] = useState<
+    Collection["id"] | undefined
+  >();
+
+  const { selectedTableIds, toggleTableIdSelection } = useSelectedTables({
+    initialValues: value.tableIds,
+    mode: "multiple",
+  });
+
+  const collectionsMap = useMemo(
+    () => _.indexBy(collections, "id"),
+    [collections],
+  );
+
+  const collectionTree = useMemo(
+    () =>
+      buildCollectionTree({
+        collections,
+        rootCollection,
+        currentUser,
+        targetModel,
+      }),
+    [collections, rootCollection, currentUser, targetModel],
+  );
+
+  const selectedItems = useMemo(() => {
+    const items: DataPickerSelectedItem[] = [];
+
+    if (selectedCollectionId) {
+      items.push({ type: "schema", id: selectedCollectionId });
+    }
+
+    const tables: DataPickerSelectedItem[] = selectedTableIds.map(id => ({
+      type: "table",
+      id,
+    }));
+
+    items.push(...tables);
+
+    return items;
+  }, [selectedCollectionId, selectedTableIds]);
+
+  const handleSelectedCollectionChange = useCallback(
+    (id: Collection["id"]) => {
+      const collection = id === "root" ? rootCollection : collectionsMap[id];
+      if (collection) {
+        setSelectedCollectionId(id);
+        const schemaId = getCollectionVirtualSchemaId(collection, {
+          isDatasets: targetModel === "model",
+        });
+        onChange({ ...value, schemaId, tableIds: [] });
+      }
+    },
+    [value, collectionsMap, rootCollection, targetModel, onChange],
+  );
+
+  const handleSelectedTablesChange = useCallback(
+    (tableId: Table["id"]) => {
+      const tableIds = toggleTableIdSelection(tableId);
+      onChange({ ...value, tableIds });
+    },
+    [value, toggleTableIdSelection, onChange],
+  );
+
+  const renderPicker = useCallback(
+    ({ schema }: { schema?: Schema } = {}) => {
+      const tables = schema?.tables
+        ? (schema.tables as VirtualTable[])
+        : undefined;
+      return (
+        <CardPickerView
+          collectionTree={collectionTree}
+          virtualTables={tables}
+          selectedItems={selectedItems}
+          targetModel={targetModel}
+          onSelectCollection={handleSelectedCollectionChange}
+          onSelectedVirtualTable={handleSelectedTablesChange}
+          onBack={onBack}
+        />
+      );
+    },
+    [
+      collectionTree,
+      selectedItems,
+      targetModel,
+      handleSelectedCollectionChange,
+      handleSelectedTablesChange,
+      onBack,
+    ],
+  );
+
+  if (!schemaId) {
+    return renderPicker();
+  }
+
+  return (
+    <Schemas.Loader id={schemaId} loadingAndErrorWrapper={false}>
```

Comment:
I believe it should be possible to use in _.compose instead of here. E.g.
```
Schema.load({
  id: (state, props) => props.value.schemaId,
  loadingAndErrorWrapper: false,
})
```

It should skip loading the entity if the `id` is not defined